src/ufuncs.c

```diff
@@ -225,8 +225,26 @@ static void Y_Y_func(char **args, npy_intp *dimensions,
 static PyUFuncGenericFunction Y_Y_funcs[1] = {&Y_Y_func};
 
 /* Define the geom, double -> geom functions (Yd_Y) */
-static void *interpolate_data[1] = {GEOSInterpolate_r};
-static void *interpolate_normalized_data[1] = {GEOSInterpolateNormalized_r};
+static void *GEOSInterpolateProtectEmpty_r(void *context, void *geom, double d) {
+    int n = GEOSGeomGetNumPoints_r(context, geom);
+    if (n < 2) {
+        /* return NULL for empty linestrings to protect against a segfault */
+        return NULL;
```

Comment:
Other option could also be to return an empty point? (although I think empty points are not very well supported?) 
In general it's always a bit difficult to reason about which of the two is best. But maybe a good rationale is that if GEOS does not return an empty geometry, we shouldn't create one.

Reply:
I don't know what I would prefer. But I think if GEOS doesn't create an empty geometry, we should do it either.

Reply:
Check out https://github.com/libgeos/geos/pull/126 , it is already solved upstream. I guess we have to keep the check for backwards compat, but do return an empty point.